kroxylicious-docs/docs/_modules/authorization/con-authorization-rules.adoc

```diff
@@ -0,0 +1,95 @@
+:_mod-docs-content-type: CONCEPT
+
+
+[id='con-authorization-rules-{context}']
+= Authorization Rules
+
+The authorization rules define who is allowed to do what.
+The basic form of a rule is as follows:
+
+[source]
+----
+<allow|deny> User with <user predicate> to <operation> <resource type> with <resource predicate>;
+----
+where:
+
+* `<allow|deny>` indicates whether to allow or deny the action.
+* `<user predicate>` matches the user performing the action.
+* `<resource type>` and `<resource predicate>` identifies the resource.
+* `<operation>` identifies the operation(s) to be performed on the resource.
+
+`<resource type>` before be declared before use.
+This is done with a `from` statement in the following form:
+
+[source]
+----
+from <package> import <element> [as <alias>][,... , <elementn> [as <aliasn>]];
+----
+where:
+
+* <package> is `io.kroxylicious.filter.authorization`
+* <element> is the `ResourceType` implementation `TopicResource`.
+* <alias> is an optional alias for the resource type.
+
+Rules files must end with the statement `otherwise deny`.
```

Comment:
what happens if rules overlap?

Reply:
was just coming to say this, this high level description of how rules are applied should probably come before we go into the individual rule syntax (and cover how deny rules must precede allow rules)

Reply:
I'm now describing rule ordering, and how rules are considered top-to-bottom with the first matching rule taking precedence.